### ISSUE TITLE
Autolinker can exceed the call stack

### ISSUE THREAD

**Issue by wojcikstefan** [gregjacobs/Autolinker.js#171]
The code below will result in `Uncaught RangeError: Maxiumum call stack exceeded`:

``` js
var test = function() {
    var t = [];
    for (var i = 0; i < 50000; i++) {
        t.push(' /&gt;&lt;br');
    };
    return t.join('');
};
a = new Autolinker();
var result = a.link(test());
```

**Comment by gregjacobs** [comment]
Fixed in 1.3.1

**Comment by charlag** [comment]
Sometimes happens to us on 1.7.1

like:

```
RangeError: Maximum call stack size exceeded
at Object.parse (https://mail.tutanota.com/main.js:1:87120)
at Autolinker.parse (https://mail.tutanota.com/main.js:1:74785)
at Autolinker.link (https://mail.tutanota.com/main.js:1:76992)
```
